### PR TITLE
Add news items for 4.11.

### PR DESCRIPTION
**Pull request by jezdez**
These are the news items that we would usually have included in PRs but since we've introduced [rever](https://regro.github.io/rever-docs/) only recently in #11039 this had to be added afterwards.

I used https://github.com/conda/conda/compare/4.10.3...master as the basis and skipped a few not as noteworthy commits. Please let me know if there is anything that doesn't make sense.

### CODE REVIEW

File: news/gh-10795-conda-ci [hunk 1]
```diff
@@ -0,0 +1,23 @@
+### Enhancements
+
+* Build Docker images periodically on GitHub Actions for the continuous
+  integration testing on Linux, storing them on GitHub Packages's registry
+  for reduced latency and cost when using Docker Hub.
+
+* Simplify the Linux GitHub actions workflows by combining used shell scripts.
```
Comment (chenghlee) [inline news/gh-10795-conda-ci:7]: Thinking we should create a new "developer" section to clearly separate user-facing enhancements (which historically has been what this section was for) and developer-/contributor-facing changes.

File: news/gh-11012-conda-canary [hunk 2]
```diff
@@ -0,0 +1,26 @@
+### Enhancements
+
+* Automatically build the conda package upon the successful merge into the
+  master branch and upload it to the conda-canary channel on anaconda.org.
+
+  To try conda out simply run:
+
+  ```
+  conda install -c conda-canary conda
```
Comment (chenghlee) [inline news/gh-11012-conda-canary:9]: Option should be `-c conda-canary/label/dev`

**Review by chenghlee — CHANGES_REQUESTED** [review]

File: news/gh-10795-conda-ci (hunk 1 above)
Comment (jezdez) [reply news/gh-10795-conda-ci:7]: Good idea, I can go through the news items and move them into a developer section.

File: news/gh-11012-conda-canary (hunk 2 above)
Comment (jezdez) [reply news/gh-11012-conda-canary:9]: Oh, huh, good catch!

File: news/gh-11040-gitpod-codespaces [hunk 3]
```diff
@@ -0,0 +1,19 @@
+### Enhancements
+
+* Enable one-click gitpod and GitHub Codespaces setup for Linux development.
```
Comment (jaimergp) [inline news/gh-11040-gitpod-codespaces:3]: We should maybe add here  a `(preview)` tag or something because we haven't used it extensively yet.

**Review by chenghlee — APPROVED** [review]

**Pull request merged**